### ISSUE TITLE
As developer I want a StUF-BG client that can retrieve StUF-mock-data

### ISSUE THREAD

**Issue by joeribekker** [maykinmedia/open-personen#41]
and later via actual StUF-BG connection (as described in #16)

The `bg0310_vraagAntwoord_prs` WSDL has 4 operations that should map to functions on the client.

The Client should be somewhat like:

```python
class Client:
  def __init__():
    # get config from solo model

  def get_gezinssituatie_op_adres_aanvrager(bsn):
    pass

  def get_kinderen_van_aanvrager(bsn):
    pass

  def get_natuurlijk_persoon(bsn):
    pass

  def get_vestiging(vestigingsNummer):
    pass
```

As example, a request that should be performed by `get_natuurlijk_persoon` (`npsLv01-prs-NatuurlijkPersoon`) :

```xml
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns="http://www.egem.nl/StUF/sector/bg/0310" xmlns:StUF="http://www.egem.nl/StUF/StUF0301">
   <soapenv:Header/>
   <soapenv:Body>
      <ns:npsLv01-prs-NatuurlijkPersoon>
         <ns:stuurgegevens>
            <StUF:berichtcode>Lv01</StUF:berichtcode>
            <StUF:zender>
               <!--Optional:-->
               <StUF:organisatie>?</StUF:organisatie>
               <StUF:applicatie>?</StUF:applicatie>
               <!--Optional:-->
               <StUF:administratie>?</StUF:administratie>
               <!--Optional:-->
               <StUF:gebruiker>?</StUF:gebruiker>
            </StUF:zender>
            <StUF:ontvanger>
               <!--Optional:-->
               <StUF:organisatie>?</StUF:organisatie>
               <StUF:applicatie>?</StUF:applicatie>
               <!--Optional:-->
               <StUF:administratie>?</StUF:administratie>
               <!--Optional:-->
               <StUF:gebruiker>?</StUF:gebruiker>
            </StUF:ontvanger>
            <!--Optional:-->
            <StUF:referentienummer>?</StUF:referentienummer>
            <StUF:tijdstipBericht>?</StUF:tijdstipBericht>
            <StUF:entiteittype>NPS</StUF:entiteittype>
         </ns:stuurgegevens>
         <ns:parameters>
            <StUF:sortering>0</StUF:sortering>
            <StUF:indicatorVervolgvraag>false</StUF:indicatorVervolgvraag>
            <StUF:maximumAantal>1</StUF:maximumAantal>
         </ns:parameters>
         <ns:gelijk StUF:entiteittype="NPS" StUF:sleutelVerzendend="?" StUF:sleutelSynchronisatie="?">
            <ns:inp.bsn StUF:noValue="?" StUF:exact="true">?</ns:inp.bsn>
         </ns:gelijk>
         <ns:scope>
            <ns:object StUF:entiteittype="NPS" StUF:sleutelVerzendend="?" StUF:sleutelSynchronisatie="?" StUF:noValue="?">
               <ns:inp.bsn StUF:noValue="?" StUF:exact="true">?</ns:inp.bsn>
               <ns:geslachtsnaam StUF:noValue="?" StUF:exact="true">?</ns:geslachtsnaam>
               <ns:voorvoegselGeslachtsnaam StUF:noValue="?" StUF:exact="true">?</ns:voorvoegselGeslachtsnaam>
               <ns:voorletters StUF:noValue="?" StUF:exact="true">?</ns:voorletters>
               <ns:voornamen StUF:noValue="?" StUF:exact="true">?</ns:voornamen>
               <ns:aanduidingNaamgebruik StUF:noValue="?" StUF:exact="true">?</ns:aanduidingNaamgebruik>
               <ns:geslachtsnaamPartner StUF:noValue="?" StUF:exact="true">?</ns:geslachtsnaamPartner>
               <ns:voorvoegselGeslachtsnaamPartner StUF:noValue="?" StUF:exact="true">?</ns:voorvoegselGeslachtsnaamPartner>
               <!--Optional:-->
               <ns:aanhefAanschrijving StUF:noValue="?" StUF:exact="true">?</ns:aanhefAanschrijving>
               <!--Optional:-->
               <ns:voornamenAanschrijving StUF:noValue="?" StUF:exact="true">?</ns:voornamenAanschrijving>
               <!--Optional:-->
               <ns:geslachtsnaamAanschrijving StUF:noValue="?" StUF:exact="true">?</ns:geslachtsnaamAanschrijving>
               <!--Optional:-->
               <ns:adellijkeTitelPredikaat StUF:noValue="?" StUF:exact="true">?</ns:adellijkeTitelPredikaat>
               <ns:geslachtsaanduiding StUF:noValue="?" StUF:exact="true">?</ns:geslachtsaanduiding>
               <ns:geboortedatum StUF:noValue="?" StUF:exact="true" StUF:indOnvolledigeDatum="V">?</ns:geboortedatum>
               <ns:inp.geboorteplaats StUF:noValue="?" StUF:exact="true">?</ns:inp.geboorteplaats>
               <ns:inp.geboorteLand StUF:noValue="?" StUF:exact="true">?</ns:inp.geboorteLand>
               <!--Optional:-->
               <ns:overlijdensdatum StUF:noValue="?" StUF:exact="true" StUF:indOnvolledigeDatum="V">?</ns:overlijdensdatum>
               <!--You have a CHOICE of the next 2 items at this level-->
               <ns:verblijfsadres>
                  <ns:aoa.identificatie StUF:noValue="?" StUF:exact="true">?</ns:aoa.identificatie>
                  <ns:wpl.identificatie StUF:noValue="?" StUF:exact="true">?</ns:wpl.identificatie>
                  <ns:wpl.woonplaatsNaam StUF:noValue="?" StUF:exact="true">?</ns:wpl.woonplaatsNaam>
                  <ns:gor.openbareRuimteNaam StUF:noValue="?" StUF:exact="true">?</ns:gor.openbareRuimteNaam>
                  <ns:gor.straatnaam StUF:noValue="?" StUF:exact="true">?</ns:gor.straatnaam>
                  <ns:aoa.postcode StUF:noValue="?" StUF:exact="true">?</ns:aoa.postcode>
                  <ns:aoa.huisnummer StUF:noValue="?" StUF:exact="true">?</ns:aoa.huisnummer>
                  <ns:aoa.huisletter StUF:noValue="?" StUF:exact="true">?</ns:aoa.huisletter>
                  <ns:aoa.huisnummertoevoeging StUF:noValue="?" StUF:exact="true">?</ns:aoa.huisnummertoevoeging>
                  <ns:inp.locatiebeschrijving StUF:noValue="?" StUF:exact="true">?</ns:inp.locatiebeschrijving>
               </ns:verblijfsadres>
               <!--Optional:-->
               <ns:sub.verblijfBuitenland>
                  <!--Optional:-->
                  <ns:lnd.landcode StUF:noValue="?" StUF:exact="true">?</ns:lnd.landcode>
                  <!--Optional:-->
                  <ns:lnd.landnaam StUF:noValue="?" StUF:exact="true">?</ns:lnd.landnaam>
                  <!--Optional:-->
                  <ns:sub.adresBuitenland1 StUF:noValue="?" StUF:exact="true">?</ns:sub.adresBuitenland1>
                  <!--Optional:-->
                  <ns:sub.adresBuitenland2 StUF:noValue="?" StUF:exact="true">?</ns:sub.adresBuitenland2>
                  <!--Optional:-->
                  <ns:sub.adresBuitenland3 StUF:noValue="?" StUF:exact="true">?</ns:sub.adresBuitenland3>
               </ns:sub.verblijfBuitenland>
               <!--Optional:-->
               <ns:sub.correspondentieAdres>
                  <!--Optional:-->
                  <ns:wpl.woonplaatsNaam StUF:noValue="?" StUF:exact="true">?</ns:wpl.woonplaatsNaam>
                  <!--Optional:-->
                  <ns:postcode StUF:noValue="?" StUF:exact="true">?</ns:postcode>
                  <!--You have a CHOICE of the next 2 items at this level-->
                  <!--Optional:-->
                  <ns:aoa.identificatie StUF:noValue="?" StUF:exact="true">?</ns:aoa.identificatie>
                  <!--Optional:-->
                  <ns:gor.openbareRuimteNaam StUF:noValue="?" StUF:exact="true">?</ns:gor.openbareRuimteNaam>
                  <!--Optional:-->
                  <ns:aoa.huisnummer StUF:noValue="?" StUF:exact="true">?</ns:aoa.huisnummer>
                  <!--Optional:-->
                  <ns:aoa.huisletter StUF:noValue="?" StUF:exact="true">?</ns:aoa.huisletter>
                  <!--Optional:-->
                  <ns:aoa.huisnummertoevoeging StUF:noValue="?" StUF:exact="true">?</ns:aoa.huisnummertoevoeging>
                  <!--Optional:-->
                  <ns:sub.postadresType StUF:noValue="?" StUF:exact="true">?</ns:sub.postadresType>
                  <!--Optional:-->
                  <ns:sub.postadresNummer StUF:noValue="?" StUF:exact="true">?</ns:sub.postadresNummer>
               </ns:sub.correspondentieAdres>
               <!--Optional:-->
               <ns:sub.telefoonnummer StUF:noValue="?" StUF:exact="true">?</ns:sub.telefoonnummer>
               <!--Optional:-->
               <ns:sub.faxnummer StUF:noValue="?" StUF:exact="true">?</ns:sub.faxnummer>
               <!--Optional:-->
               <ns:sub.emailadres StUF:noValue="?" StUF:exact="true">?</ns:sub.emailadres>
               <!--Optional:-->
               <ns:acd.code StUF:noValue="?" StUF:exact="true">?</ns:acd.code>
               <!--Optional:-->
               <StUF:extraElementen>
                  <!--1 or more repetitions:-->
                  <StUF:extraElement StUF:noValue="?" StUF:exact="true" naam="?" StUF:indOnvolledigeDatum="V">?</StUF:extraElement>
               </StUF:extraElementen>
               <!--Optional:-->
               <ns:inp.heeftAlsNationaliteit StUF:entiteittype="NPSNAT" StUF:sleutelVerzendend="?" StUF:sleutelOntvangend="?" StUF:sleutelGegevensbeheer="?" StUF:sleutelSynchronisatie="?" StUF:noValue="?">
                  <ns:gerelateerde StUF:entiteittype="NAT">
                     <!--Optional:-->
                     <ns:code StUF:noValue="?" StUF:exact="true">?</ns:code>
                     <!--Optional:-->
                     <ns:omschrijving StUF:noValue="?" StUF:exact="true">?</ns:omschrijving>
                  </ns:gerelateerde>
               </ns:inp.heeftAlsNationaliteit>
            </ns:object>
         </ns:scope>
      </ns:npsLv01-prs-NatuurlijkPersoon>
   </soapenv:Body>
</soapenv:Envelope>
```

To break it down: StUF is like SQL. It contains some meta data:

* `<soapenv:Header/>` wil be filled in later, probably with SOAP-WS stuff.
* `<StUF:zender>` and `<StUF:ontvanger>` need to be fully configured, based on some Django Solo model configuration.
* `<StUF:referentienummer>` a uuid generated on the fly, no need to log it for now
* `<StUF:tijdstipBericht>` date formatted as `%Y%m%d%H%M%S`

But it also has a TABLE and WHERE clause:

```
         <ns:gelijk StUF:entiteittype="NPS" StUF:sleutelVerzendend="?" StUF:sleutelSynchronisatie="?">
            <ns:inp.bsn StUF:noValue="?" StUF:exact="true">?</ns:inp.bsn>
         </ns:gelijk>
```

`NPS` is the "table" and the `gelijk` is a WHERE-clause, in this case, `inp.bsn` must be equal to the value provided.
You can ignore all the other element attributes for now.

The it has VALUES, called scope, to indicate which values we want:

`<ns:scope>`

Basically, we want all for now :)

You can use a template to create these requests since we only read stuff.
